internal/controllers/parser.go

```diff
@@ -834,6 +871,91 @@ func generateRoute(
 	return routeRoute, nil
 }
 
+func generateRouteWithoutCluster(corsPolicy *route.CorsPolicy, rewriteRegex *options.RewriteRegex, QoS *options.QoSOptions, websocket *bool) (*route.Route_Route, error) {
+
+	var rewritePathRegex *envoytypematcher.RegexMatchAndSubstitute
+	if rewriteRegex != nil {
+		rewritePathRegex = types.GenerateRewriteRegex(rewriteRegex.Pattern, rewriteRegex.Substitution)
+	}
+
+	var (
+		requestTimeout, requestIdleTimeout int64  = 0, 0
+		retries                            uint32 = 0
+	)
+	if QoS != nil {
+		retries = QoS.Retries
+		requestTimeout = int64(QoS.RequestTimeout)
+		requestIdleTimeout = int64(QoS.IdleTimeout)
+	}
+
+	routeRoute := &route.Route_Route{
+		Route: &route.RouteAction{},
+	}
+
+	if corsPolicy != nil {
+		routeRoute.Route.Cors = corsPolicy
+	}
+	if rewritePathRegex != nil {
+		routeRoute.Route.RegexRewrite = rewritePathRegex
+	}
+
+	if requestTimeout != 0 {
+		routeRoute.Route.Timeout = &durationpb.Duration{Seconds: requestTimeout}
+	}
+	if requestIdleTimeout != 0 {
+		routeRoute.Route.IdleTimeout = &durationpb.Duration{Seconds: requestIdleTimeout}
+	}
+
+	if retries != 0 {
+		routeRoute.Route.RetryPolicy = &route.RetryPolicy{
+			RetryOn:    "5xx",
+			NumRetries: &wrapperspb.UInt32Value{Value: retries},
+		}
+	}
+	if websocket != nil && *websocket {
+		routeRoute.Route.UpgradeConfigs = append(routeRoute.Route.UpgradeConfigs, &route.RouteAction_UpgradeConfig{UpgradeType: "websocket"})
+	}
+	// if err := routeRoute.Route.ValidateAll(); err != nil {
+	// 	return nil, fmt.Errorf("incorrect Route Action: %w", err)
+	// }
+
+	return routeRoute, nil
+}
+
+func addWeightedClusterToRoute(logger logr.Logger, routeRoute *route.Route_Route, clusterName string, weight int) *route.WeightedCluster {
+
+	weightedClusters := routeRoute.Route.GetWeightedClusters()
+	if weightedClusters != nil {
+		if weightedClusters.Clusters == nil {
+			weightedClusters.Clusters = []*route.WeightedCluster_ClusterWeight{}
+		}
+		weightedClusters.Clusters = append(weightedClusters.Clusters, &route.WeightedCluster_ClusterWeight{
+			Name:   clusterName,
+			Weight: &wrapperspb.UInt32Value{Value: uint32(weight)},
+		})
+	} else {
+		weightedClusters = &route.WeightedCluster{
+			Clusters: []*route.WeightedCluster_ClusterWeight{
+				{
+					Name:   clusterName,
+					Weight: &wrapperspb.UInt32Value{Value: uint32(weight)},
+					ResponseHeadersToAdd: []*envoy_config_core_v3.HeaderValueOption{
```

Comment:
@mbana this is what I added so we can easily from which cluster the response is coming